airbyte-webapp/src/components/ui/NextTable/NextTable.tsx

```diff
@@ -1,43 +1,72 @@
-import { ColumnDef, flexRender, useReactTable, getCoreRowModel } from "@tanstack/react-table";
+import { ColumnDef, flexRender, useReactTable, getCoreRowModel, ColumnSort } from "@tanstack/react-table";
 import classNames from "classnames";
 import { PropsWithChildren } from "react";
 
 import styles from "./NextTable.module.scss";
 
-export interface TableProps<TData> {
-  data: TData[];
-  columns: Array<ColumnDef<TData>>;
-  onClickRow?: (data: unknown) => void;
+type TData<T> = T & {
+  error?: unknown;
+};
+
+export interface TableProps<T> {
   className?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
I'm pretty sure we don't need to provide `any` type here since the goal of refactoring was to make out base table more strictly typed

Reply:
Double-checked - we can use `any` here since under the hood react-table uses the same interface